CedarJava/src/main/java/com/cedarpolicy/model/PartialAuthorizationResponse.java

```diff
@@ -3,128 +3,64 @@
 import com.cedarpolicy.Experimental;
 import com.cedarpolicy.ExperimentalFeature;
 import com.cedarpolicy.model.AuthorizationResponse.Decision;
-import com.cedarpolicy.model.AuthorizationResponse.Diagnostics;
-import com.cedarpolicy.model.slice.Policy;
 import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonProperty;
 import com.fasterxml.jackson.databind.JsonNode;
-import com.google.common.collect.ImmutableSet;
 
-import java.util.List;
+import java.util.HashMap;
+import java.util.HashSet;
 import java.util.Map;
 import java.util.Set;
 
 @Experimental(ExperimentalFeature.PARTIAL_EVALUATION)
-public abstract class PartialAuthorizationResponse {
-    private final Diagnostics diagnostics;
+public class PartialAuthorizationResponse {
+    private Decision decision;
+    private Set<String> satisfied;
+    private Set<String> errored;
+    private Set<String> may_be_determining;
+    private Set<String> must_be_determining;
+    private Map<String, JsonNode> residuals;
+    private Set<String> nontrivial_residuals;
```

Comment:
Should these be `final`?  I see that `diagnostics` was previously.  (Not a Java person and don't really know when we should/shouldn't use `final`)

Reply:
IIUC `final` just means it won't change after construction, which seems appropriate here.